File: Data Structures & Algorithms/insert-new-interval/submission-5.java
public class Solution {
    public int[][] insert(int[][] intervals, int[] newInterval) {
        List<int[]> res = new ArrayList<>();
        if (newInterval == null || newInterval.length == 0) {
            return new int[0][2];
        }
        for (int[] interval : intervals) {
            if (newInterval == null || newInterval[0] > interval[1]) {
                res.add(interval);
            } else if (newInterval[1] < interval[0]) {
                res.add(newInterval);
                res.add(interval);
                newInterval = null;
            } else {
                int min = Math.min(interval[0], newInterval[0]);
                int max = Math.max(interval[1], newInterval[1]);
                newInterval = new int[]{min, max};
            }
        }
        if (newInterval != null) res.add(newInterval);
        return res.toArray(new int[res.size()][2]);
    }
}

File: Data Structures & Algorithms/merge-intervals/submission-5.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        if len(intervals) == 1:
            return intervals
        intervals.sort(key=lambda pair:pair[0])

        res = [intervals[0]]

        for interval in intervals[1:]:
            current = res[-1]
            currentEnd = current[1]
            if interval[0] <= currentEnd:
                current[1] = max(current[1], interval[1])
            else:
                res.append(interval)
        return res                